src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,31 +44,70 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+
+        String[] parts = splitAddressIntoParts(address);
+
+        this.block = new Block(parts[ADDRESS_PARTS_BLOCK].trim());
+        this.street = new Street(parts[ADDRESS_PARTS_STREET].trim());
+        this.unit = new Unit(parts[ADDRESS_PARTS_UNIT].trim());
+        this.postalCode = new PostalCode(parts[ADDRESS_PARTS_POSTAL_CODE].trim());
     }
 
     /**
-     * Returns true if a given string is a valid person email.
+     * Splits an address into its different components that can be accessed with
+     * indices using ADDRESS_PARTS_*.
+     * 
+     * @param fullAddress to derive the parts from.
+     * @return an array of Strings, with the block, street, unit and postal
+     *         code.
+     */
+    private String[] splitAddressIntoParts(String fullAddress) {
+        return fullAddress.split(ADDRESS_SEPARATOR);
+    }
+
+    /**
+     * Returns true if a given string is a valid person address.
      */
     public static boolean isValidAddress(String test) {
         return test.matches(ADDRESS_VALIDATION_REGEX);
     }
 
     @Override
     public String toString() {
-        return value;
+        return getHumanReadableAddress(block.toString(), street.toString(), unit.toString(),
+                postalCode.toString());
+    }
+
+    /**
+     * Combines all the components into one string that is
+     * readable by human (e.g "123, some street, #45-456, 324234")
+     * 
+     * @param block number of the address
+     * @param street name of the address
+     * @param unit number of the address
+     * @param postalCode of the address
+     * @return a string with all the components combined
+     */
+    private static String getHumanReadableAddress(String block, String street, String unit,
+            String postalCode) {
```

Comment:
Why do you need this helper method to be static? Anything declared static inside a class is accessible across instances of the class. Static is not a default qualifier for a class member.